Mathlib/Data/Complex/Basic.lean

```diff
@@ -0,0 +1,1376 @@
+set_option linter.uppercaseLean3 false in
+#align complex.I_zpow_bit1 Complex.i_zpow_bit1
+
+theorem div_re (z w : ℂ) : (z / w).re = z.re * w.re / normSq w + z.im * w.im / normSq w := by
+  simp [div_eq_mul_inv, mul_assoc, sub_eq_add_neg]
+#align complex.div_re Complex.div_re
+
+theorem div_im (z w : ℂ) : (z / w).im = z.im * w.re / normSq w - z.re * w.im / normSq w := by
+  simp [div_eq_mul_inv, mul_assoc, sub_eq_add_neg, add_comm]
+#align complex.div_im Complex.div_im
+
+theorem conj_inv (x : ℂ) : conj x⁻¹ = (conj x)⁻¹ :=
+  star_inv' _
+#align complex.conj_inv Complex.conj_inv
+
+-- Porting note: removed `norm_cast`
+@[simp]
+theorem ofReal_div (r s : ℝ) : ((r / s : ℝ) : ℂ) = r / s :=
+  map_div₀ ofReal r s
+#align complex.of_real_div Complex.ofReal_div
+
+-- Porting note: removed `norm_cast`
+@[simp]
+theorem ofReal_zpow (r : ℝ) (n : ℤ) : ((r ^ n : ℝ) : ℂ) = (r : ℂ) ^ n :=
+  map_zpow₀ ofReal r n
+#align complex.of_real_zpow Complex.ofReal_zpow
+
+@[simp]
+theorem div_i (z : ℂ) : z / i = -(z * i) :=
+  (div_eq_iff_mul_eq i_ne_zero).2 <| by simp [mul_assoc]
+set_option linter.uppercaseLean3 false in
+#align complex.div_I Complex.div_i
+
+@[simp]
+theorem inv_i : i⁻¹ = -i := by
+  rw [inv_eq_one_div]
+  rw [div_i]
+  rw [one_mul]
+set_option linter.uppercaseLean3 false in
+#align complex.inv_I Complex.inv_i
+
+-- @[simp]
+/- `simp` attribute removed as linter reports this can be proved
+by `simp only [@map_inv₀]` -/
+theorem normSq_inv (z : ℂ) : normSq z⁻¹ = (normSq z)⁻¹ :=
+  map_inv₀ normSq z
+#align complex.norm_sq_inv Complex.normSq_inv
+
+-- @[simp]
+/- `simp` attribute removed as linter reports this can be proved
+by `simp only [@map_div₀]` -/
+theorem normSq_div (z w : ℂ) : normSq (z / w) = normSq z / normSq w :=
+  map_div₀ normSq z w
+#align complex.norm_sq_div Complex.normSq_div
+
+/-! ### Cast lemmas -/
+
+
+@[simp, norm_cast]
+theorem ofReal_nat_cast (n : ℕ) : ((n : ℝ) : ℂ) = n :=
+  map_natCast ofReal n
+#align complex.of_real_nat_cast Complex.ofReal_nat_cast
+
+@[simp, norm_cast]
+theorem nat_cast_re (n : ℕ) : (n : ℂ).re = n := by rw [← ofReal_nat_cast, ofReal_re]
+#align complex.nat_cast_re Complex.nat_cast_re
+
+@[simp, norm_cast]
+theorem nat_cast_im (n : ℕ) : (n : ℂ).im = 0 := by rw [← ofReal_nat_cast, ofReal_im]
+#align complex.nat_cast_im Complex.nat_cast_im
+
+@[simp, norm_cast]
+theorem ofReal_int_cast (n : ℤ) : ((n : ℝ) : ℂ) = n :=
+  map_intCast ofReal n
+#align complex.of_real_int_cast Complex.ofReal_int_cast
+
+@[simp, norm_cast]
+theorem int_cast_re (n : ℤ) : (n : ℂ).re = n := by rw [← ofReal_int_cast, ofReal_re]
+#align complex.int_cast_re Complex.int_cast_re
+
+@[simp, norm_cast]
+theorem int_cast_im (n : ℤ) : (n : ℂ).im = 0 := by rw [← ofReal_int_cast, ofReal_im]
+#align complex.int_cast_im Complex.int_cast_im
+
+@[simp, nolint synTaut]
+-- Was warned that this is a syntactic tautology.
+theorem ofReal_rat_cast (n : ℚ) : ((n : ℝ) : ℂ) = n := rfl
```

Comment:
this is labeled as a syntactic tautology because it is not the same statement as in Lean 3. To get the same statement, you would need the RHS to be `@RatCast.ratCast ℂ _ n`, but then the proof is not `rfl`. I think this is because the `Field` instance doesn't manually declare the `ratCast` bits and just uses the defaults, which have bad definitional properties. So, I'm not sure what we should do here.
```suggestion
@[simp, nolint synTaut]
-- Was warned that this is a syntactic tautology.
theorem ofReal_rat_cast (n : ℚ) : ((n : ℝ) : ℂ) = n := rfl
```

Reply:
I had changed the proof to `rfl` as the original proof did not work. But the original proof is corrected for your corrected statement :)